admin/app/indexes/TagPages.scala

```diff
@@ -82,29 +82,22 @@ object TagPages extends GuLogging {
     ("theobserver", "The Observer"),
   )
 
-  def alphaIndexKey(s: String): String = {
-    val badCharacters = """[^a-z0-9]+""".r
-
-    val maybeFirstChar = Try(badCharacters.replaceAllIn(asAscii(s).toLowerCase, "").charAt(0)).toOption
-
-    if (maybeFirstChar.isEmpty) {
-      log.error(s"Tag without alpha index, being shoved into 1-9: $s")
-    }
-
-    maybeFirstChar.filterNot(_.isDigit).map(_.toString).getOrElse("1-9")
-  }
-
-  def alphaIndexKeyForContributor(tag: Tag): String = {
-
-    /**
-      * The alphaIndexKey function looks at the first alphanumeric character in the string passed to it.
-      * Concatenating these three strings therefore allows us to use firstName and webTitle as fallbacks
-      * if lastName is None or "".
-      * */
-    val indexString =
-      tag.lastName.getOrElse("").trim.concat(tag.firstName.getOrElse("").trim).concat(tag.webTitle)
-    alphaIndexKey(indexString)
-  }
+  private val acceptedIndexCharacters: Set[Char] = ('a' to 'z').toSet ++ ('0' to '9')
+  private val numericIndexKey = "1-9"
+
+  def alphaIndexCharacter(text: String): Option[Char] = asAscii(text).toLowerCase.find(acceptedIndexCharacters)
+  def alphaIndexKey(text: String): String =
+    alphaIndexCharacter(text)
+      .map { leadChar =>
+        if (leadChar.isDigit) numericIndexKey else leadChar.toString
+      }
+      .getOrElse {
+        log.error(s"Tag without alpha index, being shoved into '$numericIndexKey': $text")
+        numericIndexKey
+      }
+
+  def alphaIndexKeyForContributor(tag: Tag): String =
+    alphaIndexKey((tag.lastName ++ tag.firstName).mkString + tag.webTitle)
```

Comment:
Here we're taking advantage of a few nice features to make the code shorter:

* `Option`s (like `tag.lastName` and `tag.firstName`) can be treated like really-short collections (that might have at most one element, or possibly zero) and those collections can be concatenated into a single collection (with at most two elements, or possibly zero) with `++`. We can use `mkString` to concatenate all the strings in that collection together into a single string (so we don't have to use `.getOrElse("")` on those `Option[String]`s).
* We can concatenate two strings in Scala using just the `+` operator, we don't need to say `concat()`!
* It's actually not necessary to perform `trim` on those strings to get the index-keying to work correctly, as the tests (now running!) validate.

I've removed the comment because I think the new tests will provide quick context of what's gone wrong if the code is broken in future.